pkg/buildkit/buildkit.go

```diff
@@ -354,19 +377,50 @@ func (c *Client) runSolve(ctx context.Context, so bkclient.SolveOpt) error {
 		return err
 	})
 
+	var size int64
+
 	eg.Go(func() error {
-		if _, err := c.bk.Solve(ctx, nil, so, ch); err != nil {
+		res, err := c.bk.Solve(ctx, nil, so, ch)
+		if err != nil {
 			return err
 		}
 
 		c.log.Info("Solve complete")
+		expresp := res.ExporterResponse
+		imageName := expresp["image.name"]
+		ref, err := name.ParseReference(imageName)
+		if err != nil {
+			return err
+		}
+		registryName := ref.Context().RegistryStr()
+		auth, err := c.ResolveAuth(registryName)
+		if err != nil {
+			return err
+		}
+		img, err := remote.Image(ref, remote.WithContext(ctx), remote.WithAuth(auth))
+		if err != nil {
+			return err
+		}
+		layers, err := img.Layers()
+		if err != nil {
+			return err
+		}
+		for _, layer := range layers {
+			compressedSize, err := layer.Size()
```

Comment:
This is the size of the compressed layer. The uncompressed layer size could probably be calculated by reading in the response from `layer.Uncompressed()`.